### PR TITLE
Added Invoke-WhiskeyNUnit3Task and tests

### LINKED ISSUE #67: Create NUnit3 Task
Create an NUnit3 task. It should be modeled after the NUnit2 task, but work with NUnit3's arguments. It should support code coverage, too.

### PR DESCRIPTION
**Pull request by jlariono**
Fixes #67

### CODE REVIEW

File: Whiskey/Tasks/Invoke-WhiskeyNUnit3Task.ps1 [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    <#
+    .SYNOPSIS
+    Runs NUnit3 tests.
+
+    .DESCRIPTION
+    The NUnit3 task runs NUnit tests. The latest version of NUnit 3 is downloaded from nuget.org for you (into a `packages` directory in your build root).
+    
+    The `Path` parameter is mandatory and must contain a list of paths, relative to your `whiskey.yml` file, to the assemblies for which NUnit tests should be run.
+
+    By default, OpenCover and ReportGenerator are also run against the results of the NUnit tests and their output is saved to the `OutputDirectory` of the build. If you wish to **only** run NUnit tests, then specify the `DisableCodeCoverage` parameter with the value of `true`.
```
Comment (splatteredbits) [inline Whiskey/Tasks/Invoke-WhiskeyNUnit3Task.ps1]: Change to `By default, OpenCover runs NUnit and gathers test code coverage, saving its report to '.output\openCover\openCover.xml'. ReporterGenerator is used to convert the OpenCover report into an HTML report, viewable to '.output\openCover\index.html'. If you wish to...`.

File: Whiskey/Tasks/Invoke-WhiskeyNUnit3Task.ps1 [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+        [Parameter(Mandatory=$true)]
+        [hashtable]
+        $TaskParameter
+    )
+    
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    $nunitPackage = 'NUnit.ConsoleRunner'
+    # Due to a bug in NuGet we can't search for and install packages with wildcards (e.g. 3.*), so we're hardcoding a version for now. See Resolve-WhiskeyNuGetPackageVersion for more details.
```
Comment (splatteredbits) [inline Whiskey/Tasks/Invoke-WhiskeyNUnit3Task.ps1]: Stupid NuGet.

File: Whiskey/Tasks/Invoke-WhiskeyNUnit3Task.ps1 [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+            ReportGeneratorVersion: 2.5.11
+            ReportGeneratorArgument:
+            - "-reporttypes: Latex"
+            - "-verbosity:Verbose"
+
+    This example will run all tests located in the `Assembly.dll` file using the default .NET framework version `4.0`. OpenCover version `4.6.519` will be run against against the NUnit test results with the `CoverageFilter` `+[Open*]* -[Open.Test]*` and with the argument `-showunvisited`. ReportGenerator version `2.5.11` will be run against the OpenCover results using the arguments `-reporttypes:Latex` and `-verbosity:Verbose`.
+    #>
+
+    [CmdletBinding()]
+    [Whiskey.Task("NUnit3",SupportsClean=$true)]
```
Comment (splatteredbits) [inline Whiskey/Tasks/Invoke-WhiskeyNUnit3Task.ps1]: It should also support initialization, e.g. `SupportsInitialize=$true`. Scott just committed this change to the NUnit 2 task. Please add this functionality, along with tests.

File: Whiskey/Tasks/Invoke-WhiskeyNUnit3Task.ps1 (hunk 1 above)
Comment (jlariono) [reply Whiskey/Tasks/Invoke-WhiskeyNUnit3Task.ps1]: Fixed.

File: Whiskey/Tasks/Invoke-WhiskeyNUnit3Task.ps1 (hunk 3 above)
Comment (jlariono) [reply Whiskey/Tasks/Invoke-WhiskeyNUnit3Task.ps1]: Fixed.

**Comment by jlariono** [comment]
Ready for re-review.

**Comment by splatteredbits** [comment]
Once the build succeeds, increment the version in Whiskey.psd1, update the release notes, and merge to master.

**Pull request merged**